src/evergreen/metrics/buildmetrics.py

```diff
@@ -200,6 +273,18 @@ def _count_task(self, task):
             if task.is_timeout():
                 self.timed_out_count += 1
 
+        if task.generated_by not in self._display_map:
+            self._display_map[task.generated_by] = task
+        else:
+            if self._display_map[task.generated_by].is_success() and not task.is_success():
```

Comment:
1. I need to add comments here to describe what is happening
2. If there are recommendations on how to implement a more idiomatic way of selecting when to override the current value in the map, let me know. Right now it's a bunch of ugly if statements.

Reply:
Do we need to handle tasks that are not generated (i.e. `generated_by` is "")?

We might be able to use an `IntEnum` to clean this up some. 

I'm thinking have an `IntEnum` for the task status. So in the `Task` class have something like this:
```
class TaskStatus(IntEnum):
    SUCCESS = 0
    FAILURE = 1
    TIMEOUT = 2
     SYSTEM_FAILURE = 3

...

def task_status():
    if self.is_success():
        return TaskStatus.SUCCESS
    if self.is_timeout():
        return TaskStatus.TIMEOUT
    if self.is_system_failure():
        return TaskStatus.SYSTEM_FAILURE
...
```

Then here we just keep a list of the tasks assigned to each display group:
```
self._display_map = defaultdict(list)
...

self._display_map[task.generated_by].append(task.task_status())
```

Then to tell the status of each display item, we can just use the max:
```
for status_list in self._display_map.values():
    status = max(status_list)
    if status == TaskStatus.SUCCESS:
        self.success_display_count +=1
    else:
        self.failure_display_count += 1
        if status == TaskStatus.SYSTEM_FAILURE:
...
```

Reply:
Yes, to the handling not generated. Let me fix that and then I'll address the rest.

Reply:
Take a look now